src/main/java/io/jenkins/plugins/gitlabbranchsource/servers/GitLabServer.java

```diff
@@ -0,0 +1,289 @@
+        return serverUrl;
+    }
+
+    /**
+     * Returns {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     *
+     * @return {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     */
+    public boolean isManageHooks() {
+        return manageHooks;
+    }
+
+    /**
+     * Returns the {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for auto-management
+     * of hooks.
+     *
+     * @return the {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for auto-management
+     * of hooks.
+     */
+    @CheckForNull
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    /**
+     * Constructor
+     *
+     * @param displayName   Optional name to use to describe the end-point.
+     * @param serverUrl     The URL of this GitLab Server
+     * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     * @param credentialsId The {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for
+     *                      auto-management of hooks.
+     * @since 1.0.5
+     */
+    @DataBoundConstructor
+    public GitLabServer(@CheckForNull String displayName, @NonNull String serverUrl, boolean manageHooks,
+                       @CheckForNull String credentialsId) {
+        this.manageHooks = manageHooks;
+        this.credentialsId = credentialsId;
+        this.serverUrl = defaultIfBlank(serverUrl, GITLAB_SERVER_URL);
+        this.name = StringUtils.isBlank(displayName)
+                ? SCMName.fromUrl(this.serverUrl, COMMON_PREFIX_HOSTNAMES)
+                : displayName;
+    }
+
+
+    /**
+     * Looks up the {@link StandardCredentials} to use for auto-management of hooks.
+     *
+     * @return the credentials or {@code null}.
+     */
+    @CheckForNull
+    public StandardCredentials credentials() {
+        return StringUtils.isBlank(credentialsId) ? null : CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        StandardCredentials.class,
+                        Jenkins.getActiveInstance(),
+                        ACL.SYSTEM,
+                        fromUri(serverUrl).build()
+                ),
+                CredentialsMatchers.allOf(
+                        AuthenticationTokens.matcher(GitLabAuth.class),
+                        withId(credentialsId)
+                )
+        );
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
+
+    /**
+     * Our descriptor.
+     */
+    @Extension
+    public static class DescriptorImpl extends Descriptor<GitLabServer> {
+
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return "GitLab Server";
```

Comment:
Is it worth making this localizable? Not sure whether most plugins do this or not

Reply:
Fixed.